src/main/java/com/maxmind/geoip2/WebServiceClient.java

```diff
@@ -296,11 +303,38 @@ public Builder proxy(ProxySelector val) {
             return this;
         }
 
+        /**
+         * @param val the custom HttpClient to use for requests. When providing a
+         *            custom HttpClient, you cannot also set connectTimeout or proxy
+         *            parameters as these should be configured on the provided client.
+         * @return Builder object
+         */
+        public Builder httpClient(HttpClient val) {
+            this.httpClient = val;
+            return this;
+        }
+
         /**
          * @return an instance of {@code WebServiceClient} created from the
          * fields set on this builder.
+         * @throws IllegalArgumentException if httpClient is provided along with
+         *                                  connectTimeout or proxy settings
          */
         public WebServiceClient build() {
+            if (httpClient != null) {
+                // Check if connectTimeout was changed from default
+                if (!connectTimeout.equals(Duration.ofSeconds(3))) {
+                    throw new IllegalArgumentException(
+                        "Cannot set both httpClient and connectTimeout. "
+                        + "Configure timeout on the provided HttpClient instead.");
+                }
+                // Check if proxy was changed from default
+                if (proxy != ProxySelector.getDefault()) {
```

Comment:
Should we be using `equals` here? Does `ProxySelector.getDefault()` always return the same reference? Perhaps it does, but I didn't dig into it. If it doesn't and we switch to `equals`, we should make sure it provides an overload that does what we want.

Another option would be to remove the default values above and set these in this method if the values are null.